src/commands/periscope/periscope.ts

```diff
@@ -0,0 +1,248 @@
+    quickPick.onDidChangeSelection(async ([{ label }]) => {
+      quickPick.hide();
+
+      await longRunning(`Generate SAS for ${clusterName} cluster.`,
+        async () => {
+          periscopeStorageInfo = await getStorageInfo(cloudResource, storageAccountNameToStorageIdMap.get(label)!);
+          await loadAndDeployPeriscope(clusterKubeConfig, periscopeStorageInfo, clusterName);
+        }
+      );
+    });
+    quickPick.show();
+  } else if (diagnosticSettings && diagnosticSettings.value!.length === 1) {
+    await longRunning(`Generate SAS for ${clusterName} cluster.`,
+      async () => {
+        periscopeStorageInfo = await getStorageInfo(cloudResource, diagnosticSettings.value![0].storageAccountId!);
+        await loadAndDeployPeriscope(clusterKubeConfig, periscopeStorageInfo, clusterName);
+      }
+    );
+  }
+}
+
+async function getStorageAccountNameAndKeyUserInputAndInstallAKSPeriscope(
+  clusterKubeConfig: string,
+  clusterName: string
+) {
+  let storageAccount = "";
+  let storageKey = "";
+
+  // "A storage account must be specified, since there isn't one in the diagnostic settings." Hence manual input- to be removed
+  await vscode.window.showInputBox({ placeHolder: 'Enter storage account name:', ignoreFocusOut: true }).then(
+    async (value) => {
+      if (value) {
+        storageAccount = value;
+        await getStorageAccountKeyUI().then((value) => storageKey = value!);
+      }
+    });
+
+  await longRunning(`Load and deploy for ${clusterName} cluster.`,
+    async () => {
+      if (storageKey && storageAccount) {
+        storageAccount = buffer.Buffer.from(storageAccount).toString('base64');
+        storageKey = buffer.Buffer.from(storageKey).toString('base64');
+        const periscopeStorageInfo = <PeriscopeStorage>{ storageName: storageAccount, storageDeploymentSas: storageKey };
+
+        await loadAndDeployPeriscope(clusterKubeConfig, periscopeStorageInfo, clusterName);
+      }
+    }
+  );
+}
+
+async function loadAndDeployPeriscope(
+  clusterKubeConfig: string,
+  periscopeStorageInfo: PeriscopeStorage | undefined,
+  clusterName: string) {
+
+  await longRunning(`Installing aks periscope for ${clusterName} cluster.`,
+    async () => {
+      const kubectl = await k8s.extension.kubectl.v1;
+
+      if (kubectl.available && periscopeStorageInfo) {
+        await inMemoryAKSPeriscopeDeplomentFileHandle(clusterKubeConfig, kubectl, periscopeStorageInfo, clusterName);
+      } else {
+        vscode.window.showInformationMessage('Associated storage to the cluster have issues.');
+      }
+    }
+  );
+}
+
+async function inMemoryAKSPeriscopeDeplomentFileHandle(
+  clusterKubeConfig: string,
+  kubectl: any,
+  periscopeStorageInfo: PeriscopeStorage,
+  clusterName: string
+): Promise<string | undefined> {
+  const https = require("https");
+  const aksPeriscopeFile = fs.createWriteStream(path.resolve(__dirname, 'aks-periscope.yaml'));
```

Comment:
I think you should create this in the temporary directory rather than the program directory.